consensus/polybft/contractsapi/contractsapi.go

```diff
@@ -0,0 +1,240 @@
+
+func (c *Commit) EncodeAbi() ([]byte, error) {
+	return StateReceiver.Abi.Methods["commit"].Encode(c)
+}
+
+func (c *Commit) DecodeAbi(buf []byte) error {
+	return decodeMethod(StateReceiver.Abi.Methods["commit"], buf, c)
+}
+
+type StateSync struct {
+	ID       *big.Int      `abi:"id"`
+	Sender   types.Address `abi:"sender"`
+	Receiver types.Address `abi:"receiver"`
+	Data     []byte        `abi:"data"`
+}
+
+var StateSyncABIType = abi.MustNewType("tuple(uint256 id,address sender,address receiver,bytes data)")
+
+func (S *StateSync) EncodeAbi() ([]byte, error) {
+	return StateSyncABIType.Encode(S)
+}
+
+func (S *StateSync) DecodeAbi(buf []byte) error {
+	return decodeStruct(StateSyncABIType, buf, &S)
+}
+
+type Execute struct {
+	Proof []types.Hash `abi:"proof"`
+	Obj   *StateSync   `abi:"obj"`
+}
+
+func (e *Execute) EncodeAbi() ([]byte, error) {
+	return StateReceiver.Abi.Methods["execute"].Encode(e)
+}
+
+func (e *Execute) DecodeAbi(buf []byte) error {
+	return decodeMethod(StateReceiver.Abi.Methods["execute"], buf, e)
+}
+
+type StateSyncResultEvent struct {
+	Counter *big.Int `abi:"counter"`
+	Status  bool     `abi:"status"`
+	Message []byte   `abi:"message"`
+}
+
+func (s *StateSyncResultEvent) ParseLog(log *ethgo.Log) error {
+	return decodeEvent(StateReceiver.Abi.Events["StateSyncResult"], log, s)
+}
+
+type NewCommitmentEvent struct {
+	StartID *big.Int   `abi:"startId"`
+	EndID   *big.Int   `abi:"endId"`
+	Root    types.Hash `abi:"root"`
+}
+
+func (n *NewCommitmentEvent) ParseLog(log *ethgo.Log) error {
+	return decodeEvent(StateReceiver.Abi.Events["NewCommitment"], log, n)
+}
+
+type Epoch struct {
+	StartBlock *big.Int   `abi:"startBlock"`
+	EndBlock   *big.Int   `abi:"endBlock"`
+	EpochRoot  types.Hash `abi:"epochRoot"`
+}
+
+var EpochABIType = abi.MustNewType("tuple(uint256 startBlock,uint256 endBlock,bytes32 epochRoot)")
+
+func (E *Epoch) EncodeAbi() ([]byte, error) {
+	return EpochABIType.Encode(E)
+}
+
+func (E *Epoch) DecodeAbi(buf []byte) error {
+	return decodeStruct(EpochABIType, buf, &E)
+}
+
+type UptimeData struct {
+	Validator    types.Address `abi:"validator"`
+	SignedBlocks *big.Int      `abi:"signedBlocks"`
+}
+
+var UptimeDataABIType = abi.MustNewType("tuple(address validator,uint256 signedBlocks)")
+
+func (U *UptimeData) EncodeAbi() ([]byte, error) {
+	return UptimeDataABIType.Encode(U)
+}
+
+func (U *UptimeData) DecodeAbi(buf []byte) error {
+	return decodeStruct(UptimeDataABIType, buf, &U)
+}
+
+type Uptime struct {
+	EpochID     *big.Int      `abi:"epochId"`
+	UptimeData  []*UptimeData `abi:"uptimeData"`
+	TotalBlocks *big.Int      `abi:"totalBlocks"`
+}
+
+var UptimeABIType = abi.MustNewType("tuple(uint256 epochId,tuple(address validator,uint256 signedBlocks)[] uptimeData,uint256 totalBlocks)")
+
+func (U *Uptime) EncodeAbi() ([]byte, error) {
+	return UptimeABIType.Encode(U)
+}
+
+func (U *Uptime) DecodeAbi(buf []byte) error {
+	return decodeStruct(UptimeABIType, buf, &U)
+}
+
+type CommitEpoch struct {
```

Comment:
Would it be a problem to suffix "Method" at the end of the methods?

Reply:
No problem. Will change it.

Reply:
Fix: https://github.com/0xPolygon/polygon-edge/pull/1062/commits/36db7ca35f9a04332c90133fc96e9c191f00608f